db/src/lib.rs

```diff
@@ -33,6 +32,14 @@ pub trait KeyValueDB: Sync + Send {
     fn batch(&self) -> Result<Self::Batch>;
 }
 
+pub trait IterableKeyValueDB: KeyValueDB {
+    fn iter<'a>(
+        &'a self,
+        col: Col,
+        from_key: &'a [u8],
+    ) -> Result<Box<Iterator<Item = KeyValueIteratorItem> + 'a>>;
```

Comment:
Why not associate type?

Reply:
I tried to use an associate type, however, it will introduce lifetime flag to the trait definition. I would like to keep simple trait design since we only have one implementation currently.